utils/gridlink_impl.c.src

```diff
@@ -169,7 +169,9 @@ cellarray_DOUBLE * gridlink_DOUBLE(const int64_t NPART,
     {
         #if defined(_OPENMP)
         int tid = omp_get_thread_num();
-        nthreads = omp_get_num_threads();  // reduce to actual number of threads, will be the same in almost all cases
+        if(tid == 0){
+            nthreads = omp_get_num_threads();  // reduce to actual number of threads, will be the same in almost all cases
+        }
```

Comment:
Not sure I follow this - `nthreads` should have the correct value on all threads - how will that be achieved? (Really, I don't understand why that call was a race condition in the first place)

Reply:
Because `nthreads` was declared outside of the parallel region, and is thus an OpenMP "shared" variable by default!

Reply:
Heh - I was wondering where that declaration was :D